src/routes/speed.svelte

```diff
@@ -1,283 +1,26 @@
-                document.getElementById("chastise").innerText = `verified ${update.city} @ ${update.latitude}, ${update.longitude} for test #${update._id}`;
-            } else {
-                // if not verified, then user gave garbage. advise to retry
-                document.getElementById("chastise").innerText = "Not a real town or zip code, check spelling and try again";
-            }
-
-        } else {
-            document.getElementById("chastise").innerText = "please separate town + state with a comma, or enter a valid 5-digit zip code";
-        }
-    };
-
-    // get most recent test results
-    function getLastTest () {    
-        let lastTest = storage.getItem('recentTest');
-        console.log(lastTest);
-        if (lastTest) {
-            let lastTestTime = parseInt(storage.getItem('recentTestDate'));
-            if (Date.now() < milliDay + lastTestTime) {
-                // last test was taken within 24 hours, show the results
-                try {
-                    testInfo = JSON.parse(lastTest);
-                    document.getElementById('result').innerHTML = `Ping: ${testInfo.ping} ms, Down: ${testInfo.downloadSpeed || "N/A"} Mbps, Up: ${testInfo.uploadSpeed || "N/A"} Mbps`;
-                    document.getElementById('done').innerHTML = `Test taken on ${new Date(lastTestTime)}`;
-                    document.getElementById('test').innerHTML = 'Click to test again';
-                } catch (error) {
-                    // if bad json parse, then just wipe localStorage
-                    storage.clear();
-                }
-            } else {
-                storage.clear();
-            }
-        }
-    };
-    onMount(getLastTest);  // onmount runs when the window comes into focus
+    var st; 
+    var testComplete = false;
+    onMount(() => {
+        st = new RuralTest();
+    })
 </script>
 
-<svelte:head>
-  <title>Speed Test</title>
-</svelte:head>
 
-    <h1>Internet Speed Test</h1>
-    {#if !finished}
-        <button id='test' class="btn btn-blue" on:click={doSpeedTest}>Click to Test</button>
-        {#if inProgress}
-            <button class="btn btn-red" id='cancel' on:click={cancelSpeedTest}>Cancel Test</button>
-        {/if}
-    {/if}
-    <p id='result'></p>
-    <p id='done'></p>
-    {#if finished}
-        <h3>Location Look Wrong?</h3>
-        <p><small>Location estimation works best on mobile devices</small></p>
-        <p><strong>Tell us where you are - city, state or zip code</strong></p>
-        <input type="text" bind:value={userLocation} placeholder="example: Barre, VT">
-        <button class="btn" on:click={fixLocation}>Update Location</button>
-        <p id="chastise"></p>
-    {/if}
\ No newline at end of file
+<h1 id='title'>New Speed Test</h1>
+<button id='test' on:click={() => {st.startTest()}}>Click to begin Speedtest</button>
+<button disabled='true' id='cancel' on:click={() => {st.abortTest()}}>Cancel Test</button>
+<p id='result'></p>
+<p id='done'></p>
```

Comment:
When cookies are disabled and after a speed test is ran. Both #result and #done IDs retain the data from that last speed test. This could be confusing to a user who declined cookies.

Previous speed test data should probably not be shown to users who have declined cookies.